oci/oci.go

```diff
@@ -227,27 +222,9 @@ func (r *Runtime) CreateContainer(c *Container, cgroupParent string) (err error)
 	childPipe.Close()
 	childStartPipe.Close()
 
-	// Move conmon to specified cgroup
-	if r.cgroupManager == SystemdCgroupsManager {
-		logrus.Infof("Running conmon under slice %s and unitName %s", cgroupParent, createUnitName("crio-conmon", c.id))
-		if err = utils.RunUnderSystemdScope(cmd.Process.Pid, cgroupParent, createUnitName("crio-conmon", c.id)); err != nil {
-			logrus.Warnf("Failed to add conmon to systemd sandbox cgroup: %v", err)
-		}
-	} else {
-		control, err := cgroups.New(cgroups.V1, cgroups.StaticPath(filepath.Join(cgroupParent, "/crio-conmon-"+c.id)), &rspec.LinuxResources{})
-		if err != nil {
-			logrus.Warnf("Failed to add conmon to cgroupfs sandbox cgroup: %v", err)
-		} else {
-			// Here we should defer a crio-connmon- cgroup hierarchy deletion, but it will
-			// always fail as conmon's pid is still there.
-			// Fortunately, kubelet takes care of deleting this for us, so the leak will
-			// only happens in corner case where one does a manual deletion of the container
-			// through e.g. runc. This should be handled by implementing a conmon monitoring
-			// routine that does the cgroup cleanup once conmon is terminated.
-			if err := control.Add(cgroups.Process{Pid: cmd.Process.Pid}); err != nil {
-				logrus.Warnf("Failed to add conmon to cgroupfs sandbox cgroup: %v", err)
-			}
-		}
+	// Platform specific container setup
+	if err := r.createContainerPlatform(c, cgroupParent, cmd.Process.Pid); err != nil {
+		logrus.Warnf("%s", err)
```

Comment:
It's surprising to me that this is non-fatal.  I realize that the previous code was non-fatal, but maybe w want to strengthen this to `return err` now that we're touching this code again.  I don't see any discussion in #385 or #812 about why we were only warning on errors before, but maybe @runcom or @mrunalp remember why they chose to go that way then?

If we stick to warnings, this should be `logrus.Warn(err)` instead of `logrus.Warnf("%s", err)`.

Reply:
i'd say no, go open a PR for that discussion. The goal of this PR is to carry existing behaviour.

Reply:
> i'd say no, go open a PR for that discussion.

Now that this PR has landed, I've opened #1518 to make these fatal errors.